jena-kafka-connector/src/main/java/org/apache/jena/kafka/common/FusekiProjector.java

```diff
@@ -286,15 +338,38 @@ protected void commitTransactionIfNeeded(Event<Bytes, RdfPayload> event) {
                     this.topicNames, this.maxTransactionDuration);
             commit();
         } else {
-            // Batch size not reached BUT we could be caught up with the Kafka topic(s) i.e. zero lag
-            Long remaining = this.source.remaining();
-            if (remaining != null && remaining == 0L) {
-                // Caught up, commit now!
-                FusekiKafka.LOG.info("[{}] Completely up to date with Kafka topic(s)", this.topicNames);
+            if (this.highLagDetected && calculateBatchSizeInBytes() > batchSizeBytes) {
+                // We are in high lag batching mode AND we've exceeded the batch size threshold
+                long sizeInBytes = calculateBatchSizeInBytes();
```

Comment:
```suggestion
            long sizeInBytes = calculateBatchSizeInBytes();
            if (this.highLagDetected && sizeInBytes > batchSizeBytes) {
                // We are in high lag batching mode AND we've exceeded the batch size threshold
```

Save us calculating it twice. If we're looking to make use of the short-circuiting of the && and not carry it out at all if high level not detected - it'll need a re-ordering of the if statements